controllers/authconfigs_reconciler.go

```diff
@@ -71,7 +71,11 @@ func (r *AuthConfigsReconciler) Reconcile(ctx context.Context, _ []controller.Re
 	modifiedAuthConfigs := []string{}
 
 	for pathID, effectivePolicy := range effectivePoliciesMap {
-		_, _, _, httpRoute, httpRouteRule, _ := kuadrantpolicymachinery.ObjectsInRequestPath(effectivePolicy.Path)
+		_, _, _, httpRoute, httpRouteRule, err := kuadrantpolicymachinery.ObjectsInRequestPath(effectivePolicy.Path)
+		if err != nil {
+			logger.Error(err, "failed to reconcile authconfig objects", "path", effectivePolicy.Path)
```

Comment:
Could we return an error type that would inform the controller of the type of failure? Failing to reconcile seems like a big deal to me? Is it safe to continue at that point or should the controller just bail and try again?

Reply:
ok so looking at https://github.com/KevFan/kuadrant-operator/blob/5d97f9aaf562314bf05e87a6fdbcb81a4328ae15/pkg/policymachinery/utils.go#L58 we return nil but also return an error explaining no HTTPRoute `NewErrInvalidPath("index 3 is not a HTTPRoute")` although I hate the message :) what does index 3 even mean
So does it mean effectively there is nothing to reconcile and so we can continue rather than we failed to reconcile?